otelconf/v0.3.0/metric_test.go

```diff
@@ -1388,3 +1394,177 @@ func TestPrometheusIPv6(t *testing.T) {
+		ctx        context.Context
+		otlpConfig *OTLPMetric
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr string
+	}{
+		{
+			name: "no TLS config",
+			args: args{
+				ctx: context.Background(),
+				otlpConfig: &OTLPMetric{
+					Protocol:    ptr("grpc"),
+					Compression: ptr("gzip"),
+					Timeout:     ptr(5000),
+					Insecure:    ptr(true),
+					Headers: []NameStringValuePair{
+						{Name: "test", Value: ptr("test1")},
+					},
+				},
+			},
+		},
+		{
+			name: "with TLS config",
+			args: args{
+				ctx: context.Background(),
+				otlpConfig: &OTLPMetric{
+					Protocol:    ptr("grpc"),
+					Compression: ptr("gzip"),
+					Timeout:     ptr(5000),
+					Certificate: ptr("testdata/server-certs/server.crt"),
+					Headers: []NameStringValuePair{
+						{Name: "test", Value: ptr("test1")},
+					},
+				},
+			},
+		},
+		{
+			name: "with TLS config and client key",
+			args: args{
+				ctx: context.Background(),
+				otlpConfig: &OTLPMetric{
+					Protocol:          ptr("grpc"),
+					Compression:       ptr("gzip"),
+					Timeout:           ptr(5000),
+					Certificate:       ptr("testdata/server-certs/server.crt"),
+					ClientKey:         ptr("testdata/client-certs/client.key"),
+					ClientCertificate: ptr("testdata/client-certs/client.crt"),
+					Headers: []NameStringValuePair{
+						{Name: "test", Value: ptr("test1")},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			n, err := net.Listen("tcp", "localhost:0")
+			require.NoError(t, err)
+
+			tt.args.otlpConfig.Endpoint = ptr(strings.ReplaceAll(n.Addr().String(), "127.0.0.1", "localhost"))
+
+			tlsMode := ""
+			if tt.args.otlpConfig.Insecure == nil || !*tt.args.otlpConfig.Insecure {
+				tlsMode = "TLS"
+			}
+			if tt.args.otlpConfig.ClientCertificate != nil && *tt.args.otlpConfig.ClientCertificate != "" {
+				tlsMode = "mTLS"
+			}
+			col, err := newGRPCMetricCollector(n, tlsMode)
+			require.NoError(t, err)
+			defer col.srv.Stop()
+
+			exporter, err := otlpGRPCMetricExporter(tt.args.ctx, tt.args.otlpConfig)
+			if tt.wantErr != "" {
+				require.Error(t, err)
+				require.Equal(t, tt.wantErr, err.Error())
+			} else {
+				require.NoError(t, err)
+			}
+
+			res, err := resource.New(context.Background())
+			require.NoError(t, err)
+
+			require.NoError(t, exporter.Export(context.Background(), &metricdata.ResourceMetrics{
+				Resource: res,
+				ScopeMetrics: []metricdata.ScopeMetrics{
+					{
+						Metrics: []metricdata.Metrics{
+							{
+								Name: "test-metric",
+								Data: metricdata.Gauge[int64]{
+									DataPoints: []metricdata.DataPoint[int64]{
+										{
+											Value: 1,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}))
+
+			// Ensure everything is flushed.
+			require.NoError(t, exporter.Shutdown(context.Background()))
+
+			// verify the reception of the sent data item
+			require.GreaterOrEqual(t, len(col.storage.data), 1)
```

Comment:
This is unnecessary. If `exporter.Export` has not returned an error it means that it succeeded. There is no need for `storage`.